src/abbfreeathome/devices/switch_sensor.py

```diff
@@ -92,12 +110,109 @@ def _refresh_state_from_datapoint(self, datapoint: dict[str, Any]) -> str:
 
             self._state = self._switch_sensor_state
             return "state"
+        if datapoint.get("pairingID") == Pairing.AL_INFO_ON_OFF.value:
+            self._led = datapoint.get("value") == "1"
+            return "led"
         return None
 
+    async def _set_led_datapoint(self, value: str):
+        """Set the led datapoint on the api."""
+        _sensor_input_id, _sensor_input_value = self.get_input_by_pairing(
+            pairing=Pairing.AL_INFO_ON_OFF
+        )
+        return await self._api.set_datapoint(
+            device_id=self.device_id,
+            channel_id=self.channel_id,
+            datapoint=_sensor_input_id,
+            value=value,
+        )
+
+    def update_device(self, datapoint_key, datapoint_value):
+        """Update the device state."""
+        super().update_device(datapoint_key, datapoint_value)
+        _callback_attribute = None
+        _io_key = datapoint_key.split("/")[-1]
+
+        if _io_key in self._inputs:
+            self._inputs[_io_key]["value"] = datapoint_value
+            _callback_attribute = self._refresh_state_from_datapoint(
+                datapoint=self._inputs[_io_key]
+            )
+
+        if _callback_attribute and self._callbacks[_callback_attribute]:
+            for callback in self._callbacks[_callback_attribute]:
+                callback()
+
+    async def refresh_state(self):
+        """Refresh the state of the device from the api."""
+        await super().refresh_state()
+
+        for _pairing in self._input_state_refresh_pairings:
+            _datapoint_id, _datapoint_value = self.get_input_by_pairing(
+                pairing=_pairing
+            )
+
+            _datapoint = (
+                await self._api.get_datapoint(
+                    device_id=self.device_id,
+                    channel_id=self.channel_id,
+                    datapoint=_datapoint_id,
+                )
+            )[0]
+
+            self._refresh_state_from_datapoint(
+                datapoint={
+                    "pairingID": _pairing.value,
+                    "value": _datapoint,
+                }
+            )
+
+    def _refresh_state_from_datapoints(self):
+        """Refresh the state of the device from the datapoints."""
+        super()._refresh_state_from_datapoints()
+
+        for _datapoint in self._inputs.values():
+            self._refresh_state_from_datapoint(_datapoint)
```

Comment:
Would it be better to add this to the `base` device class so all device could benefit from state updates based on inputs? This was originally designed for outputs so I can see this getting a bit messy if a pairing id is both in the inputs and outputs so a clear separation between the two would be better.

Maybe this is part of a bigger refactor idea, to include inputs for state updates for all devices.